packages/pigeon/run_tests.sh

```diff
@@ -86,195 +90,337 @@ test_pigeon_android() {
-cd ios
-xcodebuild \
-  -workspace Runner.xcworkspace \
-  -scheme RunnerTests \
-  -sdk iphonesimulator \
-  -destination 'platform=iOS Simulator,name=iPhone 8' \
-  test
-popd
+run_ios_e2e_tests() {
+  DARTLE_H="e2e_tests/test_objc/ios/Runner/dartle.h"
+  DARTLE_M="e2e_tests/test_objc/ios/Runner/dartle.m"
+  DARTLE_DART="e2e_tests/test_objc/lib/dartle.dart"
+  PIGEON_JAVA="e2e_tests/test_objc/android/app/src/main/java/io/flutter/plugins/Pigeon.java"
+  pub run pigeon \
+    --input pigeons/message.dart \
+    --dart_out $DARTLE_DART \
+    --objc_header_out $DARTLE_H \
+    --objc_source_out $DARTLE_M \
+    --java_out $PIGEON_JAVA
+  dartfmt -w $DARTLE_DART
+
+  pushd $PWD
+  cd e2e_tests/test_objc
+  flutter build ios -t test_driver/e2e_test.dart --simulator
+  cd ios
+  xcodebuild \
+    -workspace Runner.xcworkspace \
+    -scheme RunnerTests \
+    -sdk iphonesimulator \
+    -destination 'platform=iOS Simulator,name=iPhone 8' \
+    test
+  popd
+}
+
+run_formatter() {
+  cd ../..
+  pub global activate flutter_plugin_tools && pub global run flutter_plugin_tools format 2>/dev/null
+}
 
 ###############################################################################
-# Run the formatter on generated code.
+# main
 ###############################################################################
-cd ../..
-pub global activate flutter_plugin_tools && pub global run flutter_plugin_tools format
+should_run_dart_compilation_tests=true
+should_run_dart_unittests=true
+should_run_flutter_unittests=true
+should_run_formatter=true
+should_run_ios_e2e_tests=true
+should_run_ios_unittests=true
+should_run_java_compilation_tests=true
+should_run_mock_handler_tests=true
+should_run_objc_compilation_tests=true
+while getopts "t:l?h" opt; do
+  case $opt in
+  t)
+    should_run_dart_compilation_tests=false
+    should_run_dart_unittests=false
+    should_run_flutter_unittests=false
+    should_run_formatter=false
+    should_run_ios_e2e_tests=false
+    should_run_ios_unittests=false
+    should_run_java_compilation_tests=false
+    should_run_mock_handler_tests=false
+    should_run_objc_compilation_tests=false
+    case $OPTARG in
+    dart_compilation_tests) should_run_dart_compilation_tests=true ;;
+    dart_unittests) should_run_dart_unittests=true ;;
+    flutter_unittests) should_run_flutter_unittests=true ;;
+    ios_e2e_tests) should_run_ios_e2e_tests=true ;;
+    ios_unittests) should_run_ios_unittests=true ;;
+    java_compilation_tests) should_run_java_compilation_tests=true ;;
+    mock_handler_tests) should_run_mock_handler_tests=true ;;
+    objc_compilation_tests) should_run_objc_compilation_tests=true ;;
+    *)
+      echo "unrecognized test: $OPTARG"
+      exit 1
+      ;;
+    esac
+    ;;
+  l)
+    echo "available tests for -t:
+  dart_compilation_tests - Compilation tests on generated Dart code.
+  dart_unittests         - Unit tests on and analysis on Pigeon's Dart source code.
```

Comment:
seems like this one test is different than the rest? Should it be called something more special? 

Not clear to me whether this is a unit test against the pigeon tool implementation itself (i.e. unit tests the ast.dart etc) or it's a unit test against the pigeon schema file.

Reply:
reworded